sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3943,6 +3943,43 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34421: Resolve temporary functions and views in views with CTEs") {
```

Comment:
for the temp view, we should verify the view capture temp objects correctly, so you should also add a permanent view/function, table with the same name and see if "select * from temp_view" can return correct result

Reply:
ignore my comment, I found that is not necessary after checked your code one more time.